pandas/tests/window/test_rolling.py

```diff
@@ -1357,3 +1357,67 @@ def test_rolling_std_small_values():
     result = s.rolling(2).std()
     expected = Series([np.nan, 7.071068e-9, 7.071068e-9])
     tm.assert_series_equal(result, expected, atol=1.0e-15, rtol=1.0e-15)
+
+
+def test_rolling_mean_all_nan_window_floating_artifacts():
+    # GH#41053
+    df = DataFrame(
+        [
+            0.03,
+            0.03,
+            0.001,
+            np.NaN,
+            0.002,
+            0.008,
+            np.NaN,
+            np.NaN,
+            np.NaN,
+            np.NaN,
+            np.NaN,
+            np.NaN,
+            0.005,
+            0.2,
+        ]
+    )
+
+    expected = DataFrame(
```

Comment:
Mind parameterizing over `expected` and the slice?

Reply:
Could yes, but you make a pretty long parametrization, because most of exepected differs too

Reply:
Parametrized, got a bit better :)